core/src/main/java/io/confluent/rest/SslFactoryPemHelper.java

```diff
@@ -0,0 +1,390 @@
+      keyStore = load(isKeyStore);
+    }
+
+    @Override
+    public KeyStore get() {
+      return keyStore;
+    }
+
+    @Override
+    public char[] keyPassword() {
+      Password passwd = keyPassword != null ? keyPassword : password;
+      return passwd == null ? null : passwd.value().toCharArray();
+    }
+
+    /**
+     * Loads this keystore
+     *
+     * @return the keystore
+     * @throws KafkaException if the file could not be read or if the keystore could not be loaded
+     *                        using the specified configs (e.g. if the pass or ks type is invalid)
+     */
+    protected KeyStore load(boolean isKeyStore) {
+      try (InputStream in = Files.newInputStream(Paths.get(path))) {
+        KeyStore ks = useBcfks ? KeyStore.getInstance(type, FIPS_PROVIDER) :
+            KeyStore.getInstance(type);
+        // If a password is not set access to the truststore is still available,
+        // but integrity checking is disabled.
+        char[] passwordChars = password != null ? password.value().toCharArray() : null;
+        ks.load(in, passwordChars);
+        return ks;
+      } catch (GeneralSecurityException | IOException e) {
+        throw new KafkaException("Failed to load SSL keystore " + path + " of type " + type, e);
+      }
+    }
+
+    private Long lastModifiedMs(String path) {
+      try {
+        return Files.getLastModifiedTime(Paths.get(path)).toMillis();
+      } catch (IOException e) {
+        log.error("Modification time of key store could not be obtained: " + path, e);
+        return null;
+      }
+    }
+
+    public boolean modified() {
+      Long modifiedMs = lastModifiedMs(path);
+      return modifiedMs != null && !Objects.equals(modifiedMs, this.fileLastModifiedMs);
+    }
+
+    @Override
+    public String toString() {
+      return "SecurityStore("
+          + "path=" + path
+          + ", modificationTime="
+          + (fileLastModifiedMs == null ? null : new Date(fileLastModifiedMs)) + ")";
+    }
+  }
+
+  public static class FileBasedPemStore extends FileBasedStore {
+    public FileBasedPemStore(String path, Password keyPassword,
+                             boolean isKeyStore, boolean useBcfks) {
+      super(PEM_TYPE, path, null, keyPassword, isKeyStore, useBcfks);
+    }
+
+    @Override
+    protected KeyStore load(boolean isKeyStore) {
+      try {
+        Password storeContents = new Password(Utils.readFileAsString(path));
+        PemStore pemStore = isKeyStore
+            ? new PemStore(storeContents, storeContents, keyPassword, useBcfks)
+            : new PemStore(storeContents, useBcfks);
+        return pemStore.keyStore;
+      } catch (Exception e) {
+        log.error("Failed to load store, isKeyStore : {}, path {}", isKeyStore, path, e);
+        throw new InvalidConfigurationException("Failed to load PEM SSL keystore " + path, e);
+      }
+    }
+  }
+
+  public static class PemStore implements SecurityStore {
```

Comment:
The logic in this class and the whole file in general are very complex, can this be simplified by using classes from upstream kafka?

Reply:
Currently we are not dependant on this module in rest-utils and the code is part of ce-kafka not ak hence decided to replicate the code directly.

Reply:
> hence decided to replicate the code directly.

there is issue with this method as it will double the maintenance cost, is it possible to extract a module in ce-kafka so that rest-utils can import?

Reply:
> there is issue with this method as it will double the maintenance cost, is it possible to extract a module in ce-kafka so that rest-utils can import?

1 challenge that @smjn also alluded is that `rest-utils` is open-source, where-as ce-kafka is not.

@smjn adding to @trnguyencflt why not put the common code in a module in AK, or common or some other open source repo? And `rest-utils` and `ce-kafka` could import it. It reduces duplicate code.

Reply:
We can revisit this post FEDRAMP deadlines.

Reply:
@smjn could you please a JIRA ticket for this? Thanks 👍